common/src/main/java/software/amazon/documentdb/jdbc/common/utilities/TypeConverters.java

```diff
@@ -76,8 +76,8 @@
                 .put(short.class, new ShortConverter(0))
                 .put(String.class, new StringConverter())
                 .put(Timestamp.class, new SqlTimestampConverter())
-                .put(Byte[].class, new ArrayConverter(Byte[].class, new ByteConverter()))
-                .put(byte[].class, new ArrayConverter(byte[].class, new ByteConverter()))
+                .put(Byte[].class, new ArrayConverter(Byte[].class, new ByteConverter(), -1))
```

Comment:
Error was being thrown when value was null.  Adding -1 as default array size here makes the returned array null for that case. I think this should match up with some of the other converters having null default values.